sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+"""
+Copyright Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Parameter parsing and validation for GraphStorm SageMaker Pipeline
+"""
+
+import argparse
+import hashlib
+import json
+import logging
+import os
+from dataclasses import asdict, dataclass, fields, is_dataclass
+from typing import List
+
+
+JOB_ORDER = {
+    "gconstruct": 0,
+    "gsprocessing": 1,
+    "dist_part": 2,
+    "gb_convert": 3,
+    "train": 4,
+    "inference": 5,
+}
+
+SCRIPT_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+@dataclass
+class AWSConfig:
+    """AWS-related configuration"""
+
+    execution_role: str
+    region: str
+    graphstorm_pytorch_cpu_image_url: str
+    graphstorm_pytorch_gpu_image_url: str
+    gsprocessing_pyspark_image_url: str
+
+
+@dataclass
+class InstanceConfig:
+    """Configuration for SageMaker instances"""
+
+    train_infer_instance_count: int
+    cpu_instance_type: str
+    gpu_instance_type: str
+    graph_construction_instance_type: str
+    gsprocessing_instance_count: int
+    train_on_cpu: bool
+    volume_size_gb: int
+
+    def __post_init__(self):
+        assert (
+            self.cpu_instance_type or self.gpu_instance_type
+        ), "At least one instance type (CPU or GPU) should be specified."
+
+        if self.train_on_cpu:
+            assert (
+                self.cpu_instance_type
+            ), "Need to provide a CPU instance type when training on CPU"
+
+        if not self.graph_construction_instance_type:
+            self.graph_construction_instance_type = self.cpu_instance_type
+            logging.warning(
+                "No graph processing instance type specified, using the CPU instance type: %s",
+                self.cpu_instance_type,
+            )
+
+
+@dataclass
+class TaskConfig:
+    """Pipeline/task-level configuration"""
+
+    base_job_name: str
```

Comment:
What does base_job_name mean?
Can you add a docstr for each field? (Same for other classes.)